src/course.py

```diff
@@ -0,0 +1,92 @@
+import os
+import re
+import urllib.parse
+
+from bs4 import BeautifulSoup
+
+import globals
+from resource import Resource
+
+
+class Course:
+    def __init__(self, course_name, course_url):
+        self.name = course_name
+        self.course_url = course_url
+
+        self.course_page = globals.global_session.get(self.course_url).content
+        self.soup = BeautifulSoup(self.course_page, 'html.parser')
+        self._extract_resources()
+
+    def _extract_resources(self):
```

Comment:
Maybe we should have two separate `_extract_resources` and `extract_latest_resources` functions that return their respective map/list, instead of mutating `self.resources` and `self.latest_resources` inside the function like that.

Reply:
Any idea how to do this without having duplicates of the resource objects?

Reply:
One idea that comes to mind would be having a recent flag in the resources. So the constructor would look something along the lines of:
```
# ...
self.resources = self._extract_resources()
self.latest_resources = filter(lambda resource: resource.is_recent, self.latest_resources)
```
`self.extract_resources` would then set the flags accordingly and simply return a list of all resources.

Reply:
Good idea. I'll try something along these lines :)

Reply:
Done. Just decided to use a standard list comprehension because I think it is more comprehensible for people who didn't hear FPV ;)